app/src/main/java/com/codesoom/assignment/ToyService.java

```diff
@@ -0,0 +1,64 @@
+package com.codesoom.assignment;
+
+import com.codesoom.assignment.exceptions.ToyNotFoundException;
+
+import java.util.List;
+
+public class ToyService {
+    private ToyRepository toyRepository;
+
+    public ToyService() {
+        toyRepository = new ToyRepository();
+    }
+
+    /**
+     * 인자로 넘어온 ID를 가진 Toy를 찾아 리턴한다. 없을 경우 ToyNotFoundException을 던진다.
+     * @param id - Toy의 숫자 ID
+     * @return Toy - 인자로 넘어온 ID를 가진 Toy
+     * @throws ToyNotFoundException - 인자로 넘어온 ID를 가진 Toy가 없는 경우
+     */
+    public Toy getToyById(long id) throws ToyNotFoundException {
+        return toyRepository.findById(id).orElseThrow(() -> new ToyNotFoundException(id));
+    }
+
+    /**
+     * 인자로 넘어온 Toy를 저장하고, 저장된 Toy를 리턴한다. Toy는 호출된 순서대로 ID가 부여된다.
+     * @param toy - 저장할 Toy, ID 필드는 null인 상태이다.
+     * @return Toy - 저장된 Toy, ID 필드가 추가된채로 리턴된다.
+     */
+    public Toy register(Toy toy) {
+        Toy savedToy = toyRepository.save(toy);
+        return savedToy;
+    }
+
+    /**
+     * 저장된 모든 Toy를 리스트 형태로 리턴한다. Toy는 ID를 기준으로 정렬되어 있다.
+     * @return List<Toy> - 저장된 모든 Toy들을 리스트 형태로 리턴한다.
+     */
```

Comment:
정렬되어 있음은 이 메소드의 주제가 아닙니다. 정렬은 상세 구현에 가깝고, 메소드 안쪽을 읽어보면 이 메소드가 장담할 수 있는 내용도 아닙니다. 만약 `toyRepository.findAll()`의 구현이 변경되어 "정렬하지 않"도록 바뀐다면 이 주석은 거짓이 될 것입니다. 주석이 `toyRepository`의 구현에 의존하고 있기 때문입니다. 
```suggestion
    /**
     * 모든 Toy를 리턴한다.
     */
```

Reply:
주석과 코드의 커플링을 분리하는 일은 어렵군요 🤔  만약 이 메소드를 사용하는 유저가 '이 메소드는 리스트를 정렬해서 보내줄까?'와 같은 구현 상세가 궁금하다면 어떻게 하면 좋을까요?

Reply:
정렬된 결과를 리턴한다는 것을 어딘가에 표시하는 게 좋겠죠. 주석도 괜찮지만 Java의 표현력을 활용하는 방법을 충분히 고려할 수 있을 겁니다. 타입으로 표현을 할 수 있다는 것도 잊으면 곤란할 거에요. 즉 일급 객체를 사용한다던가...